code/modules/SCPmodular/scps/euclid/SCP-173.dm

```diff
@@ -0,0 +1,212 @@
+	a_intent = INTENT_HARM
+//	possible_a_intents = list(INTENT_HARM)
+	harm_intent_damage = 1
+	obj_damage = 500
+	melee_damage_lower = 5000
+
+	melee_damage_upper = 5000
+
+	armour_penetration = 5000
+
+
+	attacktext = "crushes"
+	attack_sound = 'sound/scpsounds/scp/spook/NeckSnap1.ogg'
+
+	spacewalk = TRUE
+
+
+
+
+	var/last_snap = 0
+	var/list/next_blinks = list()
+	var/cannot_be_seen = 1
+	animate_movement = NO_STEPS // Do not animate movement, you jump around as you're a scary statue.
+	see_in_dark = 13
+	lighting_alpha = LIGHTING_PLANE_ALPHA_MOSTLY_VISIBLE
+	vision_range = 12
+	aggro_vision_range = 12
+	sight = SEE_SELF|SEE_MOBS|SEE_OBJS|SEE_TURFS
+
+	search_objects = 1 // So that it can see through walls
+
+/mob/living/simple_animal/hostile/scp_173/say(message, bubble_type, var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null)
+	return FALSE
+
+/mob/living/simple_animal/hostile/scp_173/Initialize(mapload, var/mob/living/creator)
+	. = ..()
+	// Give spells
+	mob_spell_list += new /obj/effect/proc_holder/spell/aoe_turf/flicker_lights(src)
+	mob_spell_list += new /obj/effect/proc_holder/spell/aoe_turf/blindness(src)
+	mob_spell_list += new /obj/effect/proc_holder/spell/targeted/night_vision(src) //this is being kept just incase normal vision fucks up
+
+/mob/living/simple_animal/hostile/scp_173/proc/can_be_seen(turf/destination)
+	if(!cannot_be_seen)
+		return null
+	// Check for darkness
+	var/turf/T = get_turf(loc)
+	if(T && destination && T.lighting_object)
+		if(T.get_lumcount()<0.1 && destination.get_lumcount()<0.1) // No one can see us in the darkness, right?
+			return null
+		if(T == destination)
+			destination = null
+
+	// We aren't in darkness, loop for viewers.
+	var/list/check_list = list(src)
+	if(destination)
+		check_list += destination
+
+
+
+
+	// This loop will, at most, loop twice.
+	for(var/atom/check in check_list)
+		for(var/mob/living/M in viewers(world.view + 1, check) - src)
+			if(M.client && !M.has_unlimited_silicon_privilege)
+				if(!M.eye_blind)
+					if(next_blinks[M] == null)
+						next_blinks[M] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					if(next_blinks[M.occupant] == null)
+						next_blinks[M.occupant] = world.time+rand(15 SECONDS, 60 SECONDS)
+					return M.occupant
+	return null
+
+
+
+
+
+
+/*    VIEWLOOP
+	// This loop will, at most, loop twice.
+	for(var/atom/check in check_list)
+		for(var/mob/living/M in viewers(world.view + 1, check) - src)
+			if(M.client && !M.has_unlimited_silicon_privilege)
+				if(!M.eye_blind)
+					return M
+		for(var/obj/mecha/M in view(world.view + 1, check)) //assuming if you can see them they can see you
+			if(M.occupant && M.occupant.client)
+				if(!M.occupant.eye_blind)
+					return M.occupant
+	return null
+*/
```

Comment:
No commented out code.

Reply:
fixed in testing, next PR wont have this